### PR TITLE
[SPARK-14509][DOC] Add python CountVectorizerExample

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?

Add python CountVectorizerExample
## How was this patch tested?

manual tests

### CODE REVIEW

**Comment by zhengruifeng** [comment]
cc @jkbradley

**Comment by zhengruifeng** [comment]
cc  @holdenk Could you please take a look?

**Comment by jkbradley** [comment]
I can take a look

File: examples/src/main/python/ml/count_vectorizer_example.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    # $example on$
+    # Input data: Each row is a bag of words with a ID.
+    df = sqlContext.createDataFrame([
+        (0, "a b c".split(" ")),
+        (1, "a b b c a".split(" "))
+    ], ["id", "words"])
+
+    # fit a CountVectorizerModel from the corpus.
+    cv = CountVectorizer(inputCol="words", outputCol="features")
```
Comment (jkbradley) [inline examples/src/main/python/ml/count_vectorizer_example.py]: Pass in vocabSize=3, minDF=2 to match the Scala/Java examples

File: examples/src/main/python/ml/count_vectorizer_example.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    model = cv.fit(df)
+    result = model.transform(df)
+    for feature in result.select("id", "features").take(2):
```
Comment (jkbradley) [inline examples/src/main/python/ml/count_vectorizer_example.py]: Just write ```result.show()```

**Comment by jkbradley** [comment]
Looks good, with just a few comments

**Comment by zhengruifeng** [comment]
@jkbradley Thanks for your comments!

**Comment by jkbradley** [comment]
LGTM
Merging with master
Thanks!

**Pull request closed without merging**